components/patient-data/indexing/src/main/java/org/phenotips/data/indexing/internal/SolrPatientIndexer.java

```diff
@@ -138,4 +169,18 @@ public void reindex()
             this.logger.warn("Failed to search patients for reindexing: {}", ex.getMessage());
         }
     }
+
+    private void addGenesWithStatusToField(SolrInputDocument input, PatientData<Map<String, String>> allGenes,
+        String status, String field)
+    {
+        if (allGenes != null && allGenes.isIndexed()) {
```

Comment:
Looking at this in more detail, my understanding was only partial and I was wrong to say it wa unnecessary. That test ensures that the data is indeed a list of `Map` (rather than a single `Map` or a `Map` of `Map`s) before iterating over it.